replica/distributed_loader.cc

```diff
@@ -300,13 +301,19 @@ class table_populator {
     sharded<locator::effective_replication_map_ptr> _erms;
 
 public:
-    table_populator(global_table_ptr ptr, distributed<replica::database>& db, sstring ks, sstring cf)
+    table_populator(global_table_ptr ptr, distributed<replica::database>& db, sstring ks, sstring cf, sstring datadir)
         : _db(db)
         , _ks(std::move(ks))
         , _cf(std::move(cf))
+        , _datadir(std::move(datadir))
         , _global_table(std::move(ptr))
-        , _base_path(_global_table->dir())
-    {}
+    {
+        _base_path = database::table_dir(
+            database::keyspace_dir(_datadir, _ks),
```

Comment:
ok, not extremely elegant since the caller will have to find the right one based on its prefix, but that could work.

Reply:
No, no, I meant something different.

Right now the population loop looks like

```
    for ks in db.keyspaces {
        for dir in cfg.data_dirs {
            for table in ks.tables {
                populate_table(path(dir / ks / table.basename()))
            }
        }
    }
```

but it could look simple, like this

```
    for ks in db.keyspaces {
        for table in ks.tables {
            for dir in table.datadirs() {
                populate_table(path(dir))
            }
        }
    }
```